src/optimizer/join_order/relation_manager.cpp

```diff
@@ -0,0 +1,351 @@
+#include "duckdb/optimizer/join_order/relation_manager.hpp"
+
+#include "duckdb/common/assert.hpp"
+#include "duckdb/common/enums/join_type.hpp"
+#include "duckdb/common/printer.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/optimizer/join_order/relation_statistics_helper.hpp"
+#include "duckdb/planner/operator/list.hpp"
+
+namespace duckdb {
+
+const vector<RelationStats> RelationManager::GetRelationStats() {
+	vector<RelationStats> ret;
+	for (idx_t i = 0; i < relations.size(); i++) {
+		ret.push_back(relations[i]->stats);
+	}
+	return ret;
+}
+
+vector<unique_ptr<SingleJoinRelation>> RelationManager::GetRelations() {
+	return std::move(relations);
+}
+
+idx_t RelationManager::NumRelations() {
+	return relations.size();
+}
+
+struct DistinctCount;
```

Comment:
I don't think that this forward declaration is necessary. This struct should be included in this file. Might need a forward declaration in the header, but that's it